client/src/featureform/serving.py

```diff
@@ -1020,6 +1035,51 @@ def dataframe(self) -> pd.DataFrame:
             self._dataframe.rename(columns={cols.label: "label"}, inplace=True)
             return self._dataframe
 
+    def __get_file_format_and_location(self, location: str) -> str:
+        try:
+            file_format = FileFormat.get_format(location)
+        except Exception:
+            file_format = "parquet"
+
+        is_individual_part_file = location.split("/")[-1].startswith("part-")
```

Comment:
Can you split these parsing bits into their own functions with comments on what its doing?